### DISCUSSION TITLE
How to install and use the NT8 ATSQuadroStrategyBase

### DISCUSSION THREAD

**Discussion by MicroTrendsTom** [MicroTrendsLtd/NinjaTrader8#61] [Show and tell]
A)Note: Also you will need to remove the prior files from ATSQuadroBase within NinjaTrader 8 strategies folder and start from fresh 
B) After Installation please see Discussion for how to prepare to use NT8 Code Editor: https://github.com/MicroTrendsLtd/NinjaTrader8/discussions/65#discussion-4123348

Summary 
Basic GitHub skills are required to  get the source or use a zip file and extract,  Browse to and Select source folder. Copy. Browse to target folder paste over target - yes to copy.  

1) get the code from GitHub. Clone or Code Zip file  for the work in progress:
e.g: https://github.com/MicroTrendsLtd/NinjaTrader8/archive/refs/heads/main.zip
or get a release for the stable prior: https://github.com/MicroTrendsLtd/NinjaTrader8/releases

![image](https://user-images.githubusercontent.com/24366913/154948426-c6d6b517-14ae-486c-8f73-cb36a92b2387.png)
2) Open the downloaded folder system copy the NinjaTrader 8 folder structure over the top of the local documents NinjaTrader 8 folders
![image](https://user-images.githubusercontent.com/24366913/154948885-e1f4e67b-8ca7-4405-8c60-07202752b363.png)
Copy & Paste and click yes to replace existing files etc.
![image](https://user-images.githubusercontent.com/24366913/154948739-e10e84ee-dd54-48a4-823a-7ea6da115852.png)

3) Start NT8 - open the NT8 Code Editor and compile the code
4) For Visual Studio open from the code editor or double click the NinjaTrader.Custom.cpProj
![image](https://user-images.githubusercontent.com/24366913/154849373-d115fb4c-2f8b-4a95-a82c-53160ae16c88.png)

**Comment by TLable** [comment]
Where do all the files inside the GUI folder in the Compersses Zip go, and does the Properties Folder need to be unpacked and where is it's location in the NT8 dir. I had added the GUI folder files inside the NT8 dir bin>custom which returned Compiler Errors when starting up.

**Reply by MicroTrendsTom** [reply]
please see below for another way to do it using the main zip method.

https://github.com/MicroTrendsLtd/NinjaTrader8/discussions/61#discussioncomment-2891823

**Reply by MicroTrendsTom** [reply]
From extracting the Zip file and browsing to the "NinjaTrader 8" folder
This is then simply an operating system folder copy and paste action over the top of the existing Docuemts/NinjaTrader folders etc. 
All the files go to the correct place by virtue of the folder structure

**Comment by MicroTrendsTom** [comment]
For when you use zip file:  https://github.com/MicroTrendsLtd/NinjaTrader8/archive/refs/heads/main.zip
A)Note: Also you will need to remove the prior files from ATSQuadroBase within NinjaTrader 8 strategies folder and start from fresh 
B) After Installation please see Discussion for how to prepare to use NT8 Code Editor:

**Summary of actions**
Get and download the zip file and extract, Browse to and Select source folder. Copy. Browse to target folder paste over target - yes to copy.

https://github.com/MicroTrendsLtd/NinjaTrader8/discussions/65#discussion-4123348

Please follow these exact steps and if you get stuck please tell us which step in a new issue: 
1)Download  https://github.com/MicroTrendsLtd/NinjaTrader8/archive/refs/heads/main.zip to your PC downloads folder
![image](https://user-images.githubusercontent.com/24366913/172197649-72edba6a-966b-4c6e-bb83-80d01607d1ab.png)

2)Use the mouse right click the  zip file and "extract all / extract to here"
![image](https://user-images.githubusercontent.com/24366913/172197800-e8654495-1159-4240-9cef-95dc76122734.png)

3) From the folder NinjaTrader8-Main  - browse to the "NinjaTrader 8" folder  ignore all other folders this is all you need.
![image](https://user-images.githubusercontent.com/24366913/172198024-384bb7ae-0c39-426b-b04d-feacea047f37.png)

4) Copy the "NinjaTrader 8" folder - paste/merge over the top of your existing NinjaTrader 8 the same as shown above
![image](https://user-images.githubusercontent.com/24366913/172198461-61838972-b093-4dd6-b945-6e3b7bc25a2d.png)

5)Paste into the parent directory to merge/copy over the top eg copy into "Documents" so the folder goes to 
"Documents/NinjaTrader 8"
![image](https://user-images.githubusercontent.com/24366913/172198575-3ac086b5-c118-4eae-a1cc-d796de260e3b.png)

6) browse into the folder system and view the files e.g: "\Documents\NinjaTrader 8\bin\Custom\Strategies"
![image](https://user-images.githubusercontent.com/24366913/172198898-ccb28f0e-776f-4c83-aeec-37d248ab7721.png)

**Comment by MicroTrendsTom** [comment]
Any issues or errors please create a new issue so this is easy to read and follow for others :-)
https://github.com/MicroTrendsLtd/NinjaTrader8/issues
